src/onnx/parse_mxfixneuron.cpp

```diff
@@ -129,32 +130,38 @@ struct parse_mxfixneuron : op_parser<parse_mxfixneuron>
             input,
             block_scales_ins); // output is float_type
 
-        // ravel tensor to 1D for handling possible odd number of elements for packing
+        // packing axis set to fastest dimension
         auto quantized_shape     = q_ins->get_shape();
-        std::size_t num_elements = quantized_shape.elements();
-        auto ravel_ins =
-            info.add_instruction(make_op("reshape", {{"dims", {num_elements}}}), q_ins);
-        bool odd_num_elem = (num_elements % 2 == 1);
-        if(odd_num_elem)
+        const auto& qs_strides   = quantized_shape.strides();
+        if(qs_strides.empty())
         {
-            // pad one element at end if odd number of elements
-            ravel_ins = info.add_instruction(make_op("pad", {{"pads", {0, 1}}}), ravel_ins);
+            MIGRAPHX_THROW("MXFixNeuron: quantized_shape has no strides");
         }
-        auto pack_ins =
-            info.add_instruction(make_op("pack_fp4"), ravel_ins); // output is fp4x2_type
-        auto unpack_ins =
-            info.add_instruction(make_op("unpack_fp4"), pack_ins); // output is fp8e4m3fn_type
-        if(odd_num_elem)
+        int fast_axis =
+            std::min_element(qs_strides.cbegin(), qs_strides.cend()) - qs_strides.cbegin();
+        bool odd_fast_axis = (quantized_shape.lens().at(fast_axis) % 2 == 1);
```

Comment:
nitpick: consider assigning `quantized_shape.lens().at(fast_axis)` to some variable called fast_axis_len and use it again when adding the unpack ins